src/main/java/com/aws/greengrass/mqtt/bridge/clients/LocalMqtt5Client.java

```diff
@@ -0,0 +1,391 @@
+        try {
+            this.client = new Mqtt5Client(mqtt5ClientOptions);
+        } catch (CrtRuntimeException e) {
+            throw new MQTTClientException("Unable to create an MQTT5 client", e);
+        }
+    }
+
+    private MqttMessage publishToMqttMessage (Publish publish) {
+        return MqttMessage.builder()
+                .userProperties(publish.getUserProperties())
+                .messageExpiryIntervalSeconds(publish.getMessageExpiryIntervalSeconds())
+                .payload(publish.getPayload())
+                .contentType(publish.getContentType())
+                .correlationData(publish.getCorrelationData())
+                .retain(publish.isRetain())
+                .payloadFormat(publish.getPayloadFormat())
+                .responseTopic(publish.getResponseTopic())
+                .topic(publish.getTopic())
+                .build();
+    }
+
+    private List<UserProperty> convertUserProperty (List<com.aws.greengrass.mqttclient.v5.UserProperty> userProperties) {
+        return userProperties.stream()
+                .map(obj -> new UserProperty(obj.getKey(), obj.getValue()))
+                .collect(Collectors.toList());
+    }
+
+    private void reset() {
+        // TODO: Follow up with Client Cert changes and reinstalling bridge
+    }
+
+    @Override
+    public void publish(MqttMessage message) throws MessageClientException {
+        if(client.getIsConnected()) {
+
+            PublishPacket publishPacket = new PublishPacket.PublishPacketBuilder()
+                    .withPayload(message.getPayload())
+                    .withQOS(QOS)
+                    .withTopic(message.getTopic())
+                    .withRetain(message.isRetain())
+                    .withPayloadFormat(PublishPacket.PayloadFormatIndicator.getEnumValueFromInteger(message.getPayloadFormat().getValue()))
+                    .withContentType(message.getContentType())
+                    .withCorrelationData(message.getCorrelationData())
+                    .withResponseTopic(message.getResponseTopic())
+                    .withUserProperties(convertUserProperty(message.getUserProperties()))
+                    .withMessageExpiryIntervalSeconds(message.getMessageExpiryIntervalSeconds())
+                    .build();
+            LOGGER.atDebug().kv("topic", message.getTopic()).kv("message", message).log("Publishing message to MQTT "
+                    + "topic");
+            //TODO: In the future, think about returning future
+            CompletableFuture<PublishResult> publishFuture = client.publish(publishPacket);
+            //TODO: get puback
+            try {
+                PublishResult pubReturn = publishFuture.get();
+                PubAckPacket pubAckPacket = pubReturn.getResultPubAck();
+                //TODO: Fix logging
+                LOGGER.atDebug().kv("reason", pubAckPacket.getReasonString()).log("while publishing");
+            } catch (InterruptedException | ExecutionException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)